fixtures/tests/test_fixture.py

```diff
@@ -185,7 +203,54 @@ def test_addDetail(self):
             self.assertEqual({}, fixture.getDetails())
             fixture.addDetail('foo', 'content')
         # Cleanup clears the details too.
-        self.assertEqual({}, fixture.getDetails())
+        self.assertRaises(TypeError, fixture.getDetails)
+
+    def test_setUp_subclassed(self):
+        # Even though its no longer recommended, we need to be sure that
+        # overriding setUp and calling super().setUp still works.
+        class Subclass(fixtures.Fixture):
+            def setUp(self):
+                super(Subclass, self).setUp()
+                self.fred = 1
+                self.addCleanup(setattr, self, 'fred', 2)
+        with Subclass() as f:
+            self.assertEqual(1, f.fred)
+        self.assertEqual(2, f.fred)
+
+    def test__setUp(self):
+        class Subclass(fixtures.Fixture):
+            def _setUp(self):
+                self.fred = 1
+                self.addCleanup(setattr, self, 'fred', 2)
+        with Subclass() as f:
+            self.assertEqual(1, f.fred)
+        self.assertEqual(2, f.fred)
+
+    def test__setUp_fails(self):
+        class Subclass(fixtures.Fixture):
+            def _setUp(self):
+                self.addDetail('log', text_content('stuff'))
+                1/0
+        f = Subclass()
+        e = self.assertRaises(fixtures.MultipleExceptions, f.setUp)
```

Comment:
In this and the following test, there are a lot of assertions and it takes a chunk of time to parse them. I think it's worth summarizing what these test expect to happen in comments.